be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -299,11 +322,12 @@ uint64_t SegmentWriter::estimate_segment_size() {
     return size;
 }
 
-Status SegmentWriter::finalize(uint64_t* segment_file_size, uint64_t* index_size) {
-    // check disk capacity
-    if (_data_dir != nullptr && _data_dir->reach_capacity_limit((int64_t)estimate_segment_size())) {
-        return Status::InternalError("disk {} exceed capacity limit.", _data_dir->path_hash());
+Status SegmentWriter::finalize_columns(uint64_t* index_size) {
+    if (_has_key) {
```

Comment:
Please add comment for usage. e.g. appendkey finalize_columns  or we can add a method finalize_key_columns and it takes row_count as a argument.